modules/checklists/checklists.php

```diff
@@ -955,12 +956,15 @@ public function save_global_checklist()
                 return;
             }
 
-            if (!wp_verify_nonce($_POST['_wpnonce'], 'ppch-global-checklists')) {
+            if (!wp_verify_nonce(sanitize_text_field($_POST['_wpnonce']), 'ppch-global-checklists')) {
                 return;
             }
 
+            // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
             $new_options = $_POST['publishpress_checklists_checklists_options'];
 
+            //sanitize checklists options
+            $this->sanitize_checklists_options($new_options);
```

Comment:
The method `sanitize_checklists_options` receives the variable `$new_options` but is not reassigning its value based on the return value of that method. Considering it is an array, I don't think PHP will automatically pass it as reference. If the method returns the sanitized value, we should have this:

```
$new_options = $this->sanitize_checklists_options($new_options);
```

Reply:
updated

Reply:
Thanks, this has been updated.